### ISSUE TITLE
Retain function name, when using observer wrapper with an arrow function?

### ISSUE THREAD

**Issue by RosenTomov** [mobxjs/mobx-react-lite#242]
Currently, I'm using 'ErrorBoundary' class component (implements componentDidCatch etc.) that wraps up the hooks, since they don't support error catching yet.
```
<ErrorBoundary>
  <App>
</ErrorBoundary>
```

Is there some tricky way to retain or set the function name, when using observer wrapper with an arrow function?

```
const App = observer(() => { 
...
throw new Error('test')
...
});
```
Results in
```
The above error occurred in the <wrappedComponent> component:
    in wrappedComponent
    in ErrorBoundary
```
   
  
I've tried settings displayName and defaultProps.name with no luck.
   
   
The only workaround so far is to convert the arrow function to a named one:
```
const App = observer(function App() { 
....
});
```

Which results in:
```
The above error occurred in the <App> component:
    in App
    in ErrorBoundary
```

**Comment by mweststrate** [comment]
Use App.displayname = "App"

Op wo 20 nov. 2019 11:29 schreef RosenTomov <notifications@github.com>:

**Comment by RosenTomov** [comment]
It didn't work, I've also tried with "defaultProps.name".

**Comment by mweststrate** [comment]
sorry, should be displayName I think. It is a standard React feature.

**Comment by RosenTomov** [comment]
Yeah, that's what I mean 
```
App.displayName = 'App';
```

**Comment by mweststrate** [comment]
That should work, can you create a reproduction code sandbox?

**Comment by mweststrate** [comment]
For further details see https://mobx.js.org/best/pitfalls.html#the-display-name-of-react-components-is-not-set

**Comment by FredyC** [comment]
We can hopefully solve this better eventually with a macro #203

**Comment by RosenTomov** [comment]
Sorry for the delay, here's an example
https://codesandbox.io/s/mobx-react-lite-example-su99m?fontsize=14&hidenavigation=1&theme=dark

Check the console.
```
The above error occurred in the <wrappedComponent> component:
    in wrappedComponent (at src/index.js:10)
    in TestWrapper (at src/index.js:9)
```
even with `App.displayName = "App";`
  
  
And with named function (`const App = observer(function App() {`)
```
The above error occurred in the <App> component:
    in App (at src/index.js:10)
    in TestWrapper (at src/index.js:9)
```

**Comment by FredyC** [comment]
Sorry for the delay here. I've tried to debug this and I don't think there is anything we can do about it.

Using `App.displayName = 'App'` doesn't work because the error message is generated by React which looks at `App.type.displayName` internal object. That also applies to React Devtools.

There are two ways to tackle this correctly and the name will be correctly displayed in both cases.

```js
const App = () => {
  return <div className="App">TEST</div>;
};

export default observer(App);
```

```js
const App = observer(function App() {
  return <div className="App">TEST</div>;
});

export default App
```

Closing this for now. Let us know if you have further questions.
